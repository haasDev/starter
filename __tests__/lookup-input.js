import React from 'react';
import LookupInput from '../src/lookup-input';
import {mount} from 'enzyme';
import snapshot from './snapshot';

describe('LookupInput', () => {
  it('should render', () => {
    snapshot(<LookupInput img="search" />);
  });

  it('should call onChange fn when text is updated with entered text', () => {
    const onChange = jest.fn();
    const wrapper = mount(<LookupInput img="search" onChange={onChange} />);
    const input = wrapper.find('.lookup-input-input');
    input.simulate('change', {target: {value: 'foo'}});
    expect(onChange)
      .toHaveBeenCalledWith(
        expect.objectContaining({target: {value: 'foo'}})
      );
  });

  it('should call onClick fn when the icon is clicked', () => {
    const onClick = jest.fn();
    const wrapper = mount(<LookupInput img="search" onSubmit={onClick} />);
    const icon = wrapper.find('.lookup-input-icon');
    icon.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should call onClick fn when enter key is pressed', () => {
    const onEnterPress = jest.fn();
    const wrapper = mount(<LookupInput img="search" onSubmit={onEnterPress} />);
    const input = wrapper.find('.lookup-input-input');
    input.simulate('keyPress', {which: 13});
    expect(onEnterPress).toHaveBeenCalled();
  });
});

