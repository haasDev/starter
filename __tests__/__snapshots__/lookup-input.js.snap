exports[`LookupInput should render 1`] = `
<div
  className="lookup-input-container"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }>
  <input
    className="form-control lookup-input-input"
    onChange={undefined}
    onKeyPress={[Function]}
    style={
      Object {
        "flex": "9 9 0px",
        "paddingRight": "32px",
        "width": "90%",
      }
    }
    type="text" />
  <i
    className="lookup-input-icon fa fa-search"
    onClick={undefined}
    style={
      Object {
        "flex": "1 1 0px",
        "left": "-20px",
        "position": "relative",
      }
    } />
</div>
`;
